xds/internal/xdsclient/client.go

```diff
@@ -94,12 +107,16 @@ func (c *clientImpl) Close() {
 	// Note that Close needs to check for nils even if some of them are always
 	// set in the constructor. This is because the constructor defers Close() in
 	// error cases, and the fields might not be set when the error happens.
-	if c.controller != nil {
-		c.controller.Close()
-	}
-	if c.pubsub != nil {
-		c.pubsub.Close()
+
+	c.authorityMu.Lock()
+	authorities := c.authorityPerConfig
```

Comment:
Since this is at close time, does it really matter that much that we store these in a local variable, release the lock and then call close on the individual entries. Can't we skip the local vars, and directly call `close()` on the authorities with the lock being held?

Reply:
Done.

I think I did it this way because `idleCache.Clear` calls the cleanup callback, which needs the `authorityMu`. But it seems that's no long the case.